### PR TITLE
Update membership to use new rbac endpoints

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Btw, started working on this.  So far, seems a clean transition to use the new RBAC endpoints on the membership page.  Wish I would have had some e2e test coverage, but the unit tests are fairly decent. 

Depends on [Common 262](https://github.com/openshift/origin-web-common/pull/262)

I'll just have to add the GVR to the common/constants.

Re: [future Trello card](https://trello.com/c/R1ninKHf)

@jwforres @spadgett 
fyi @enj

### CODE REVIEW

**Comment by jwforres** [comment]
So the objects look the same? If there is no impact to create/update/delete of users in roles then I'm ok updating the membership page.

I'm more concerned that it looks like selfsubjectrulesreviews is dropped in the new API? @enj whats the appropriate replacement for that?

**Comment by benjaminapetersen** [comment]
Gonna keep [WIP] on this till i test a bit more, but atm has been 100% compat.

dunno about `selfsubjectrulesreviews`.  @enj?

**Comment by enj** [comment]
`oc get --raw /apis/authorization.k8s.io/v1 | jq '.resources[] | .name'`

```
"localsubjectaccessreviews"
"selfsubjectaccessreviews"
"selfsubjectrulesreviews"
"subjectaccessreviews"
```

@benjaminapetersen in terms of differences in the objects, everything is captured by https://github.com/openshift/origin/blob/master/pkg/authorization/apis/authorization/rbacconversion/conversion.go

**Comment by jwforres** [comment]
Ok not sure why the list in bens trello card was missing it

On Dec 1, 2017 5:59 PM, "Mo Khan" <notifications@github.com> wrote:

oc get --raw /apis/authorization.k8s.io/v1 | jq '.resources[] | .name'

"localsubjectaccessreviews"
"selfsubjectaccessreviews"
"selfsubjectrulesreviews"
"subjectaccessreviews"

@benjaminapetersen <https://github.com/benjaminapetersen> in terms of
differences in the objects, everything is captured by
https://github.com/openshift/origin/blob/master/pkg/authorization/apis/
authorization/rbacconversion/conversion.go

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin-web-console/pull/2550#issuecomment-348636690>,
or mute the thread

**Comment by benjaminapetersen** [comment]
Rebased.

Not ready until we cut another release of common.

**Comment by spadgett** [comment]
I don't see a reason to wait if it's working. We should update the Trello card so QE regression tests.

**Comment by benjaminapetersen** [comment]
Sounds good. I just moved the Trello to in-progress & labeled.

**Comment by benjaminapetersen** [comment]
https://github.com/openshift/origin-web-common/pull/262

**Comment by benjaminapetersen** [comment]
https://github.com/openshift/origin-web-common/pull/263

File: app/scripts/controllers/membership.js [hunk 1]
```diff
@@ -25,6 +26,12 @@ angular
       var annotation = $filter('annotation');
       var canI = $filter('canI');
 
+      var serviceAccountVersion = APIService.getPreferredVersion('serviceaccounts');
+      var roleBindingsVersion = APIService.getPreferredVersion('rolebindings');
+
+      console.log('service accounts', serviceAccountVersion);
```
Comment (jwforres) [inline app/scripts/controllers/membership.js]: remove console.log stmts here and in the other files

**Comment by benjaminapetersen** [comment]
rebased, & updated the `update` and `create` fns to use `objectToResourceGroupVersion`.

**Comment by spadgett** [comment]
/lgtm

Is this still WIP?

**Comment by spadgett** [comment]
dist mismatch

**Comment by benjaminapetersen** [comment]
rebased

**Pull request merged**